source/mir/math/stat.d

```diff
@@ -420,6 +424,176 @@ unittest
     assert(float.max.repeat(3).hmean!(double, "fast").approxEqual(float.max));
 }
 
+private template centerImpl(F)
+{
+    import mir.ndslice.slice: Slice, SliceKind;
+
+    @safe pure nothrow @nogc
+    auto centerImpl(Iterator, size_t N, SliceKind kind)(
+        Slice!(Iterator, N, kind) slice, F average)
+    {
+        return slice - average;
+    }
+}
+
+/++
+Centers `r`, which must be a finite iterable.
+
+By default, `r` is centered by the mean. A custom function may also be provided
+using `centralTendency`.
+
+Returns:
+    The elements in the range r with the average subtracted from them.
++/
+
+/// ditto
+template center(alias centralTendency = mean!(Summation.appropriate))
+{
+    /++
+    Params:
+        r = range
+    +/
+    @safe pure nothrow
+    auto center(Range)(Range r)
+        if (isIterable!Range)
+    {
+        return centerImpl!(sumType!Range)(r, centralTendency(r));
+    }
+    
+    /++
+    Params:
+        r = range
+        average = average
+    +/
+    @safe pure nothrow
+    auto center(Range, T)(Range r, out T average)
+        if (isIterable!Range)
+    {
+        average = centralTendency(r);
+        return centerImpl!T(r, average);
+    }
+}
+
+/// Center vector
+version(mir_test)
+@safe pure nothrow
+unittest
+{
+    import mir.ndslice.slice: sliced;
+    import mir.algorithm.iteration: all;
+    import mir.math.common: approxEqual;
+
+    auto x = [0.0, 1, 2, 3, 4, 5].sliced;
+    auto result = [-2.5, -1.5, -0.5, 0.5, 1.5, 2.5].sliced;
+    assert(x.center.all!approxEqual(result));
+    
+    //Can also output average
+    double avg;
+    auto y = x.center(avg);
+    assert(avg == 2.5);
+}
+
+/// Can center using different averages
+version(mir_test)
+@safe pure nothrow
+unittest
+{
+    import mir.ndslice.slice: sliced;
+    import mir.algorithm.iteration: all;
+    import mir.math.common: approxEqual;
+
+    auto x = [1.0, 2, 3, 4, 5, 6].sliced;
```

Comment:
would be good to add two inner centered overload that accepts an array or a range with `asSlice` attribute, converts them to Slice and then call the first one overload (check how it is done for `topology.map`)

Reply:
Addressed some of your points, but still need to do this. Will try to get it done later today.

Reply:
@9il I was trying to find something to use for a unittest for the asSlice overload, but mir.series didn't work because sum doesn't work for it. Is there something else I can use to test the asSlice overload?

```
unittest {
    import mir.series;
    import mir.math.sum;

    auto index = [0, 1, 2, 3, 4, 5];
    auto data = [1.0, 2, 3, 4, 5, 6];
    auto s = index.series(data);
    assert(s.sum == 21); // fails
}
```